apps/aecore/src/aec_tx_processor.erl

```diff
@@ -898,6 +906,27 @@ contract_create({OwnerPubkey, Amount, Deposit, GasLimit, GasPrice,
                                        RollbackS, S5)
     end.
 
+%%%-------------------------------------------------------------------
+
+tx_event_op(Name) ->
+    {tx_event, Name}.
+
+tx_event(Name, #state{tx_env = Env} = S) ->
+    case aetx_env:signed_tx(Env) of
```

Comment:
I have seen this code in another place...

Reply:
Yes, in `aesc_utils.erl:add_event/3`. I've refactored it now.